backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoicecandidates/impl/InvoicesRestControllerImpl.java

```diff
@@ -157,6 +159,35 @@ public InvoicesRestControllerImpl(
 		return ResponseEntity.notFound().build();
 	}
 
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Invoice info retrieved"),
+			@ApiResponse(code = 401, message = "You are not authorized"),
+			@ApiResponse(code = 404, message = "No Invoice found"),
+			@ApiResponse(code = 422, message = "An error happened during Invoice info retrieval"),
+	})
+	@GetMapping(path = "{invoiceId}/invoiceInfo", produces = "application/json")
+	@Override
+	public ResponseEntity<?> getInvoiceInfo(
+			@ApiParam(required = true, value = "metasfreshId of the invoice")
+			@PathVariable("invoiceId") final int invoiceRecordId)
+	{
+		final InvoiceId invoiceId = InvoiceId.ofRepoIdOrNull(invoiceRecordId);
+		if (invoiceId == null)
+		{
+			return ResponseEntity.notFound().build();
+		}
+
+		try
+		{
+			final JSONInvoiceInfoResponse invoiceInfo = invoiceService.getInvoiceInfo(invoiceId, Env.getAD_Language());
+			return ResponseEntity.ok(invoiceInfo);
+		}
+		catch (final Exception e)
+		{
+			return ResponseEntity.unprocessableEntity().body(e.getLocalizedMessage());
```

Comment:
pls use JsonError class that we already have

Reply:
i.e. de.metas.common.rest_api.JsonError

Reply:
done